### PR TITLE
Add command cooldowns

### PR DESCRIPTION
**Pull request by Blocksnmore**
## About

This command adds the `commandOnCooldown` event, along with a `cooldown` property on both `CommandClient` and `Command`

## Status

- [x] These changes have been tested against Discord API or do not contain API change.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

File: src/commands/client.ts [hunk 1]
```diff
@@ -404,6 +415,25 @@ export class CommandClient extends Client implements CommandClientOptions {
...
     }
 
+    const userCooldowns = this.lastUsed.get(msg.author.id) ?? {
+      global: 0,
+      commands: {}
+    }
+    const commandCanBeUsedAt =
+      (userCooldowns.commands[command.name] ?? 0) +
+      ((command.cooldown ?? 0) as number)
+    const globalCooldown = userCooldowns.global + this.cooldown
```
Comment (Helloyunho) [inline src/commands/client.ts:425]: shouldn't one of them override the cooldown or am I reading the code wrong?

Comment (Helloyunho) [reply src/commands/client.ts:425]: but it adds the command cooldown with the global cooldown...

Comment (nodgear) [reply src/commands/client.ts:425]: my 2 cents:
global cooldown should prevent any user from running command `x` before the cooldown expires.

For example: resource heavy commands and/or for minigames like having a command that can only be redeemed by one person every `y` hours.

```ts
/**
* Defines a cooldown in miliseconds, preventing the same user from running the command within the cooldown period
**/
command.cooldown?: number
/**
* Defines a global cooldown in miliseconds, preventing any user from running the command within the cooldown period
**/
command.globalCooldown?:number
```

Comment (Blocksnmore) [reply src/commands/client.ts]: > my 2 cents: global cooldown should prevent any user from running command `x` before the cooldown expires.
> 
> For example: resource heavy commands and/or for minigames like having a command that can only be redeemed by one person every `y` hours.
> 
> ```ts
> /**
> * Defines a cooldown in miliseconds, preventing the same user from running the command within the cooldown period
> **/
> command.cooldown?: number
> /**
> * Defines a global cooldown in miliseconds, preventing any user from running the command within the cooldown period
> **/
> command.globalCooldown?:number
> ```

I've added the global cooldowns and also renamed the previous cooldown to hopefully reduce confusion in regards to what it applies to.

**Review by Helloyunho — APPROVED** [review]
lgtm, finally got what you were trying to do

**Pull request merged**